.ci/docker.run

```diff
@@ -69,9 +69,15 @@ elif [ "$CC" == "clang" ]; then
   ../configure --enable-unit --enable-self-generated-certificate  --enable-integration --with-maxloglevel=none --with-crypto=$WITH_CRYPTO $CONFIGURE_OPTIONS
 else
   if [ "$WITH_TCTI" == "mssim" ]; then
-    ../configure --with-sanitizer=undefined,address --disable-tcti-swtpm --enable-unit --enable-self-generated-certificate --enable-integration --with-crypto=$WITH_CRYPTO $CONFIGURE_OPTIONS
+    if [ ldconfig -p 2>/dev/null| grep libasan > /dev/null && ldconfig -p 2>/dev/null| grep libubsan > /dev/null ]; then
+        SANITIZER_OPTION="--with-sanitizer=undefined,address"
+    fi
```

Comment:
Why not move this block up to something like line 65 and just set the variable there rather than repeating the code block to set it and then just keep the references to the option in the configure line?

Reply:
@williamcroberts If we will include an alpine Image we cannot run these two tests with the sanitizer enabled. Currently the option sanitizer can be used even when the two libs are not available.

Reply:
Yeah I get that, but why duplicate the code to set the variable. This patch should be functionally equivelent:

```diff
diff --git a/.ci/docker.run b/.ci/docker.run
index 9d6914d3bc5f..e835b584298d 100755
--- a/.ci/docker.run
+++ b/.ci/docker.run
@@ -63,20 +63,18 @@ if [ "$CC" == "gcc" ]; then
   export CONFIGURE_OPTIONS="$CONFIGURE_OPTIONS --enable-code-coverage";
 fi
 
+if [ ldconfig -p 2>/dev/null| grep libasan > /dev/null && ldconfig -p 2>/dev/null| grep libubsan > /dev/null ]; then
+    SANITIZER_OPTION="--with-sanitizer=undefined,address"
+fi
+
 if [ "$SCANBUILD" == "yes" ]; then
   scan-build --status-bugs ../configure --enable-unit --enable-self-generated-certificate --enable-integration --with-crypto=$WITH_CRYPTO $CONFIGURE_OPTIONS
 elif [ "$CC" == "clang" ]; then
   ../configure --enable-unit --enable-self-generated-certificate  --enable-integration --with-maxloglevel=none --with-crypto=$WITH_CRYPTO $CONFIGURE_OPTIONS
 else
   if [ "$WITH_TCTI" == "mssim" ]; then
-    if [ ldconfig -p 2>/dev/null| grep libasan > /dev/null && ldconfig -p 2>/dev/null| grep libubsan > /dev/null ]; then
-        SANITIZER_OPTION="--with-sanitizer=undefined,address"
-    fi
     ../configure $SANITIZER_OPTION --disable-tcti-swtpm --enable-unit --enable-self-generated-certificate --enable-integration --with-crypto=$WITH_CRYPTO $CONFIGURE_OPTIONS
   else
-    if [ ldconfig -p 2>/dev/null| grep libasan > /dev/null && ldconfig -p 2>/dev/null| grep libubsan > /dev/null ]; then
-      SANITIZER_OPTION="--with-sanitizer=undefined,address"
-    fi
     ../configure $SANITIZER_OPTION --with-maxloglevel=none --enable-debug=yes --enable-unit --enable-self-generated-certificate --enable-integration --with-crypto=$WITH_CRYPTO $CONFIGURE_OPTIONS
   fi
 fi
```